packages/next-intl/src/extractor/ExtractionCompiler.test.tsx

```diff
@@ -2840,6 +2838,121 @@ describe('custom format', () => {
   });
 });
 
+describe('folder operations', () => {
+  function createCompiler() {
+    return new ExtractionCompiler(
+      {
+        srcPath: './src',
+        sourceLocale: 'en',
+        messages: {
+          path: './messages',
+          format: 'json',
+          locales: 'infer'
+        }
+      },
+      {
+        isDevelopment: true,
+        projectRoot: '/project'
+      }
+    );
+  }
+
+  it('removes messages when a folder is deleted', async () => {
+    filesystem.project.src = {
+      components: {
+        'Button.tsx': `
+          import {useExtracted} from 'next-intl';
+          function Button() {
+            const t = useExtracted();
+            return <div>{t('Click me')}</div>;
+          }
+        `
+      }
+    } as any;
+
+    filesystem.project.messages = {
+      'en.json': '{"OpKKos": "Click me"}'
+    };
+
+    using compiler = createCompiler();
+    await compiler.extractAll();
+    await waitForWriteFileCalls(1);
+
+    // Simulate deleting the directory
+    delete (filesystem.project.src as any).components;
+    fileTimestamps.delete('/project/src/components/Button.tsx');
+
+    const callback = parcelWatcherCallbacks.get('/project/src');
+    if (!callback) throw new Error('No watcher found for /project/src');
```

Comment:
@cursoragent You can just assert non-null (`!`) here since we are in a test (check also other tests)